src/meepgeom.cpp

```diff
@@ -1249,54 +1321,82 @@ void geom_epsilon::fallback_chi1inv_row(meep::component c, double chi1inv_row[3]
     }
     return;
   }
-
   number esterr;
-  integer errflag, n;
+  integer errflag;
   number xmin[3], xmax[3];
-  vector3 gvmin, gvmax;
-  gvmin = vec_to_vector3(v.get_min_corner());
-  gvmax = vec_to_vector3(v.get_max_corner());
-  xmin[0] = gvmin.x;
-  xmax[0] = gvmax.x;
-  if (dim == meep::Dcyl) {
-    xmin[1] = gvmin.z;
-    xmin[2] = gvmin.y;
-    xmax[1] = gvmax.z;
-    xmax[2] = gvmax.y;
+  double meps, minveps;
+
+  if (md->which_subclass == material_data::MATERIAL_GRID) {
+    matgrid_volavg mgva;
+    mgva.dim = v.dim;
+    mgva.ugrad_abs = meep::abs(gradient);
+    mgva.uval = uval;
+    mgva.rad = v.diameter()/2;
+    mgva.beta = md->beta;
+    mgva.eta = md->eta;
+    mgva.med1_eps_diag = md->medium_1.epsilon_diag;
+    mgva.med2_eps_diag = md->medium_2.epsilon_diag;
+    xmin[0] = -v.diameter()/2;
+    xmin[1] = 0;
+    xmin[2] = 0;
+    xmax[0] = v.diameter()/2;
+    xmax[1] = 0;
+    xmax[2] = 0;
```

Comment:
`cadaptive_integration` will only look at at the first component of `xmin` and `xmax` since your integral has dimension 1.